### PR TITLE
Run sphinx-lint on Misc/NEWS.d/next/

### PR DESCRIPTION
**Pull request by hugovk**
Follow on from https://github.com/python/cpython/pull/105750.

Before that PR, we used to run sphinx-lint on `Doc/` and `Misc/NEWS.d/next/`

After, we only ran sphinx-lint on `Doc/`. I think leaving out `Misc/NEWS.d/next/` was an oversight on my part, let's add it back.

In the meantime, a handful of NEWS warnings had slipped through, this PR also fixes them.

### CODE REVIEW

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

Although I do not see a difference in some files. Why are they marked as different?

**Comment by hugovk** [comment]
They were "No newline at end of file":

```
Misc/NEWS.d/next/Library/2023-07-14-16-54-13.gh-issue-106752.BT1Yxw.rst:1: found an unbalanced inline literal markup. (unbalanced-inline-literals-delimiters)
Misc/NEWS.d/next/Library/2023-07-14-16-54-13.gh-issue-106752.BT1Yxw.rst:1: default role used (hint: for inline literals, use double backticks) (default-role)
Misc/NEWS.d/next/Library/2020-05-03-00-33-15.bpo-18319.faPTlx.rst:1: default role used (hint: for inline literals, use double backticks) (default-role)
Misc/NEWS.d/next/Library/2020-05-03-00-33-15.bpo-18319.faPTlx.rst:2: default role used (hint: for inline literals, use double backticks) (default-role)
Misc/NEWS.d/next/Core and Builtins/2023-06-24-10-34-27.gh-issue-105775.OqjoGV.rst:1: No newline at end of file. (missing-final-newline)
Misc/NEWS.d/next/Library/2023-08-17-14-45-25.gh-issue-105736.NJsH7r.rst:2: default role used (hint: for inline literals, use double backticks) (default-role)
Misc/NEWS.d/next/Library/2023-08-17-14-45-25.gh-issue-105736.NJsH7r.rst:3: default role used (hint: for inline literals, use double backticks) (default-role)
Misc/NEWS.d/next/Core and Builtins/2023-07-04-04-50-14.gh-issue-100288.yNQ1ez.rst:3: default role used (hint: for inline literals, use double backticks) (default-role)
Misc/NEWS.d/next/Library/2023-06-30-16-42-44.gh-issue-106263.tk-t93.rst:2: No newline at end of file. (missing-final-newline)
Misc/NEWS.d/next/Build/2023-07-28-18-17-33.gh-issue-106881.U3Ezdq.rst:1: default role used (hint: for inline literals, use double backticks) (default-role)
Misc/NEWS.d/next/Core and Builtins/2023-07-27-11-47-29.gh-issue-104432.oGHF-z.rst:2: default role used (hint: for inline literals, use double backticks) (default-role)
Misc/NEWS.d/next/Library/2023-08-07-14-12-07.gh-issue-107715.238r2f.rst:1: default role used (hint: for inline literals, use double backticks) (default-role)
Misc/NEWS.d/next/Library/2023-07-16-10-40-34.gh-issue-106789.NvyE3C.rst:1: role use a single backtick, double backtick found. (role-with-double-backticks)
Misc/NEWS.d/next/Library/2023-07-16-10-40-34.gh-issue-106789.NvyE3C.rst:1: role use a single backtick, double backtick found. (role-with-double-backticks)
Misc/NEWS.d/next/Library/2023-06-25-12-28-55.gh-issue-106075.W7tMRb.rst:1: default role used (hint: for inline literals, use double backticks) (default-role)
Misc/NEWS.d/next/Core and Builtins/2023-07-27-11-18-04.gh-issue-106078.WEy2Yn.rst:1: No newline at end of file. (missing-final-newline)
Misc/NEWS.d/next/Library/2023-07-22-21-57-34.gh-issue-107089.Dnget2.rst:2: No newline at end of file. (missing-final-newline)
```

https://github.com/hugovk/cpython/actions/runs/5925848833/job/16066091943

File: Misc/NEWS.d/next/Library/2023-07-14-16-54-13.gh-issue-106752.BT1Yxw.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Fixed several bugs in zipfile.Path, including: in ``Path.match`, Windows
+Fixed several bugs in zipfile.Path, including: in ``Path.match``, Windows
```
Comment (AlexWaygood) [inline Misc/NEWS.d/next/Library/2023-07-14-16-54-13.gh-issue-106752.BT1Yxw.rst]: We could possibly make this an actual reference:

```suggestion
Fixed several bugs in zipfile.Path, including: in :meth:`zipfile.Path.match`, Windows
```

**Comment by hugovk** [comment]
The 3.12 backport will only need the first commit, I'll backport it manually.

File: Misc/NEWS.d/next/Library/2023-08-17-14-45-25.gh-issue-105736.NJsH7r.rst [hunk 2]
```diff
@@ -1,3 +1,3 @@
-Harmonized the pure Python version of OrderedDict with the C version. Now,
-both versions set up their internal state in `__new__`.  Formerly, the pure
-Python version did the set up in `__init__`.
+Harmonized the pure Python version of ``OrderedDict`` with the C version. Now,
```
Comment (AlexWaygood) [inline Misc/NEWS.d/next/Library/2023-08-17-14-45-25.gh-issue-105736.NJsH7r.rst]:
```suggestion
Harmonized the pure Python version of :class:`~collections.OrderedDict` with the C version. Now,
```

File: Misc/NEWS.d/next/Library/2023-08-07-14-12-07.gh-issue-107715.238r2f.rst [hunk 3]
```diff
@@ -1 +1 @@
-Fix `doctest.DocTestFinder.find` in presence of class names with special characters. Patch by Gertjan van Zwieten.
+Fix ``doctest.DocTestFinder.find`` in presence of class names with special characters. Patch by Gertjan van Zwieten.
```
Comment (AlexWaygood) [inline Misc/NEWS.d/next/Library/2023-08-07-14-12-07.gh-issue-107715.238r2f.rst]:
```suggestion
Fix :meth:`doctest.DocTestFinder.find` in presence of class names with special
characters. Patch by Gertjan van Zwieten.
```

**Review by AlexWaygood — APPROVED** [review]
Looks good, a few optional suggestions:

File: Misc/NEWS.d/next/Library/2023-08-07-14-12-07.gh-issue-107715.238r2f.rst (hunk 3 above)
Comment (hugovk) [reply Misc/NEWS.d/next/Library/2023-08-07-14-12-07.gh-issue-107715.238r2f.rst]: (For some reason GitHub didn't show this in the batch view)

**Pull request merged**